include/spdlog/sinks/periodic_file_sink.h

```diff
@@ -0,0 +1,231 @@
+// Copyright(c) 2015-present, Gabi Melman & spdlog contributors.
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+//File: minute_file_sink to support every 1 min file rotation.
+//example driver code:
+// auto duration{std::chrono::minutes{1}};
+//auto logger = spdlog::periodic_logger_mt("periodic_logger", "logs/min-log.txt",duration,60,false);
+
+#pragma once
+
+#include <spdlog/common.h>
+#include <spdlog/details/file_helper.h>
+#include <spdlog/details/null_mutex.h>
+#include <spdlog/fmt/fmt.h>
+#include <spdlog/sinks/base_sink.h>
+#include <spdlog/details/os.h>
+#include <spdlog/details/circular_q.h>
+#include <spdlog/details/synchronous_factory.h>
+
+#include <chrono>
+#include <cstdio>
+#include <ctime>
+#include <mutex>
+#include <string>
+
+namespace spdlog {
+namespace sinks {
+
+/*
+ * Generator of Minute log file names in format basename.YYYY-MM-DD-HH-MM.ext
+ */
+struct periodic_filename_calculator
+{
+    // Create filename for the form basename.YYYY-MM-DD-H-M
+    static filename_t calc_filename(const filename_t &filename, const tm &now_tm)
+    {
+        filename_t basename, ext;
+        std::tie(basename, ext) = details::file_helper::split_by_extension(filename);
+        return fmt_lib::format(SPDLOG_FMT_STRING(SPDLOG_FILENAME_T("{}_{:04d}-{:02d}-{:02d}-{:02d}_{:02d}{}")), basename, now_tm.tm_year + 1900, now_tm.tm_mon + 1,
+            now_tm.tm_mday, now_tm.tm_hour,now_tm.tm_min,ext);
+    }
+};
+
+/*
+ * Rotating file sink based on time(minutes).
+ * If truncate != false , the created file will be truncated.
+ * If max_files > 0, retain only the last max_files and delete previous.
+ */
+template<typename Mutex, typename FileNameCalc = periodic_filename_calculator>
+class periodic_file_sink final : public base_sink<Mutex>
+{
+public:
+   // create every periodic file sink which rotates on given time
+    /*periodic_file_sink(
+        filename_t base_filename, bool truncate = false, uint16_t max_files = 0,
+        std::chrono::duration<int> rotation_period = 0, const file_event_handlers &event_handlers = {})
+        
+       */
+    periodic_file_sink(
+       filename_t base_filename, 
+       std::chrono::minutes rotation_interval,
+       uint16_t max_files,
+       bool truncate_first = false,       
+       const file_event_handlers &event_handlers = {})
+        : base_filename_(std::move(base_filename))
+        , file_helper_{event_handlers}
+        , truncate_(truncate_first)
+        , max_files_(max_files)
```

Comment:
It would be great if the max file size was also configurable
like 20 * 1024 *1024 （20MB）
like [rotating_file_sink](https://github.com/gabime/spdlog/blob/2312489bdc304886b4a2eebc8f7f8dc0c4de688f/include/spdlog/sinks/rotating_file_sink.h#L47C14-L47C14)

Reply:
@kegechen 

Thanks for the suggestion,I will look in to that.I have some issues on running the test case as I mention above.If someone can help to fix it I can quickly close the PR.

Reply:
Can't help, unfortunately I'm not Not familiar with the test case either
[Catch2 tutorial](https://github.com/catchorg/Catch2/blob/devel/docs/tutorial.md#top) may help

Reply:
by the way, to many commits in your pr. should have rebase first

Reply:
@kegechen 

Sorry for the delay due to some urgent tasks,I'm going to start working on this again.Will update the progress